### PR TITLE
feat: Write tests for burn operations #12726

### PR DESCRIPTION
**Pull request by stefan-stefanooov**
Write E2E suites for HSCS operations affected by the security model
Token Burn
-->

**Related issue(s)**:
https://github.com/hashgraph/hedera-services/issues/12726
Fixes #
https://github.com/hashgraph/hedera-services/issues/12726

### CODE REVIEW

**Review by fchitakova — DISMISSED** [review]

**Review by petreze — COMMENTED** [review]
In `ContractKeysHTSV1SecurityModelSuite` we have `burnTokenWithFullPrefixAndPartialPrefixKeys()` test that is not updated for v2. Is it on purpose?

**Comment by stefan-stefanooov** [comment]
> In `ContractKeysHTSV1SecurityModelSuite` we have `burnTokenWithFullPrefixAndPartialPrefixKeys()` test that is not updated for v2. Is it on purpose?

Hey, @petreze 
I think it is.... the new name is `V2SecurityBurnTokenWithFullPrefixAndPartialPrefixKeys`.
Am I missing something ?

**Review by petreze — DISMISSED** [review]
LGTM

File: hedera-node/test-clients/src/main/java/com/hedera/services/bdd/suites/contract/precompile/ContractBurnHTSV2SecurityModelSuite.java [hunk 1]
```diff
@@ -669,6 +694,421 @@ final HapiSpec V2Security039FungibleTokenWithDelegateContractKeyCanNotBurnFromDe
...
+    final HapiSpec V2SecurityBurnTokenWithFullPrefixAndPartialPrefixKeys() {
+        final var firstBurnTxn = "firstBurnTxn";
+        final var secondBurnTxn = "secondBurnTxn";
+        final var amount = 99L;
+        final AtomicLong fungibleNum = new AtomicLong();
+
+        return propertyPreservingHapiSpec("burnTokenWithFullPrefixAndPartialPrefixKeys")
+                .preserving(CONTRACTS_ALLOW_SYSTEM_USE_OF_HAPI_SIGS, CONTRACTS_MAX_NUM_WITH_HAPI_SIGS_ACCESS)
+                .given(
+                        overridingTwo(
```
[the remark this answers is not in the document]
Comment (stefan-stefanooov) [reply hedera-node/test-clients/src/main/java/com/hedera/services/bdd/suites/contract/precompile/ContractBurnHTSV2SecurityModelSuite.java]: Removed

**Review by petreze — APPROVED** [review]

**Review by petreze — APPROVED** [review]
LGTM

**Review by kimbor — APPROVED** [review]
lgtm!

**Pull request merged**